shared/js/background/message-handlers.js

```diff
@@ -145,12 +145,27 @@ export function getTopBlockedByPages (options) {
 }
 
 // Click to load interactions
-export async function initClickToLoad (config, sender) {
+export async function initClickToLoad (contentScopeConfig, sender) {
     await settings.ready()
     const tab = tabManager.get({ tabId: sender.tab.id })
 
     // Remove first-party entries.
     await startup.ready()
+
+    /**
+     * The Click to Load configuration is in the process of being migrated, but it is currently split
+     * between tdsStorage.ClickToLoadConfig (aka social_ctp_configuration.json), the block list (aka tds.json),
+     * and content-scope-scripts/features/click-to-play.js.
+     * Here we take care to remove any entries from the content-scope-scripts configuration (aka contentScopeConfig)
+     * that aren't also found in tdsStorage.ClickToLoadConfig - otherwise the feature can end up in an inconsistent state.
+     */
+    const config = {}
```

Comment:
Not sure I follow the logic here.  The legacy config should be deprecated and no longer being updated, so should not override the embedded one.  There's no reason to really update the legacy one at this point?

Reply:
From what I got when testing this, we still seem to use `tdsStorage.ClickToLoadConfig` to define which domains we block and surrogate rules. And this configuration is still being pulled from `social_ctp_configuration.json`. Without this, what was happening with YT CTL is that we were using the embedded configuration and adding placeholders to the page without actually blocking any of the YT requests. So I added this filtering here to make both configurations match. Am I missing something here along the way and is there a better way to link YT CTL in C-S-S and the list of domains we block?

Reply:
Update: @ladamski and I synced and he explained that `social_ctp_configuration.json` will be deprecated soon in favor of using rules coming from `tds`. So by the time YT CTL releases, live configuration won't be used anymore for this and we can sync landing the changes to enable YT CTL in both `tds` and `content-scope-scripts` in the same release. So this filtering won't be necessary anymore.

Reply:
[]()

Reply:
> # - [ ] - **_@~~``~~ _**